File: app/src/main/java/com/github/overpass/gather/auth/register/UsernameValidator.java
package com.github.overpass.gather.auth.register;

import java.util.regex.Pattern;

public class UsernameValidator {

    public boolean isUsernameValid(String username) {
        return Pattern.compile("[A-Za-z0-9_]+")
                .matcher(username)
                .matches();
    }
}

File: app/src/main/java/by/overpass/gather/model/validator/UsernameValidator.kt
package by.overpass.gather.model.validator

import java.util.regex.Pattern
import javax.inject.Inject

class UsernameValidator @Inject constructor() : Validator<String> {

    override fun isValid(username: String): Boolean {
        return Pattern.compile("[A-Za-z0-9_]+")
                .matcher(username)
                .matches()
    }
}
